packages/spl-token-bonding/src/bondingHierarchy.ts

```diff
@@ -72,6 +77,35 @@ export class BondingHierarchy {
     return found;
   }
 
+  highestOrUndefined(one: PublicKey, two: PublicKey): PublicKey | undefined {
+    return this.toArray().find(
+      (hierarchy) =>
+        hierarchy.tokenBonding.baseMint.equals(
+          sanitizeSolMint(one, this.wrappedSolMint)
+        ) ||
+        hierarchy.tokenBonding.baseMint.equals(
+          sanitizeSolMint(two, this.wrappedSolMint)
+        )
+    )?.tokenBonding?.baseMint;
+  }
+
+  highest(one: PublicKey, two: PublicKey, childEntangledMint?: PublicKey | undefined, parentEntangledMint?: PublicKey | undefined): PublicKey {
```

Comment:
Is this something that would be better with composition? I.e. you create an entangled bonding hierarchy that calls out to the bonding hierarchy but makes these assumptions? Or via inheritance? Feels odd to have them as args.

Reply:
Yeah probably better for separation of concerns, will do